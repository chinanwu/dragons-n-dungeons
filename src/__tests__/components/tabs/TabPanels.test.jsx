import { shallow } from 'enzyme';
import React from 'react';

import {
  mapStateToProps,
  TabPanels,
} from '../../../components/tabs/TabPanels.jsx';

jest.unmock('../../../components/tabs/TabPanels.jsx');

describe('TabPanels component', () => {
  describe('rendering', () => {
    it('renders TabPanels', () => {
      const wrapper = shallow(<TabPanels />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders TabPanels with empty panels', () => {
      const wrapper = shallow(<TabPanels panels={[]} />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });

    it('renders TabPanels with panels', () => {
      const wrapper = shallow(
        <TabPanels
          panels={['Some things', 'Other content']}
          activeIndex={0}
          theme="light"
        />
      );
      expect(wrapper.getElement()).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('returned mapped properties', () => {
      const theme = 'light';
      const state = { theme };
      expect(mapStateToProps({ theme: state })).toEqual({ theme: theme });
    });
  });
});
