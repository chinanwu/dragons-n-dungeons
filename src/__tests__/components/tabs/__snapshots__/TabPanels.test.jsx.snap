// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPanels component rendering renders TabPanels 1`] = `
<div
  className="TabPanels"
/>
`;

exports[`TabPanels component rendering renders TabPanels with empty panels 1`] = `
<div
  className="TabPanels"
/>
`;

exports[`TabPanels component rendering renders TabPanels with panels 1`] = `
<div
  className="TabPanels"
>
  <div
    aria-labelledby="tabButton__0"
    id="tabPanel__0"
  >
    Some things
  </div>
  <div
    aria-labelledby="tabButton__1"
    id="tabPanel__1"
  >
    Other content
  </div>
</div>
`;
